sdgym/datasets.py

```diff
@@ -45,9 +47,24 @@ def _get_dataset_path(dataset, datasets_path, bucket=None, aws_key=None, aws_sec
     return dataset_path
 
 
-def load_dataset(dataset, datasets_path=None, bucket=None, aws_key=None, aws_secret=None):
+def load_dataset(dataset, datasets_path=None, bucket=None, aws_key=None, aws_secret=None,
+                 max_columns=None):
     dataset_path = _get_dataset_path(dataset, datasets_path, bucket, aws_key, aws_secret)
-    metadata = Metadata(str(dataset_path / 'metadata.json'))
+    metadata_file = open(str(dataset_path / 'metadata.json'))
+    metadata_content = json.load(metadata_file)
+    if max_columns:
+        for table in metadata_content['tables']:
+            fields = metadata_content['tables'][table]['fields']
+            if len(fields) > max_columns:
+                fields = dict(itertools.islice(fields.items(), max_columns))
+                metadata_content['tables'][table]['fields'] = fields
+            if 'structure' in metadata_content['tables'][table]:
+                metadata_content['tables'][table]['structure']['structure'] = \
```

Comment:
I would avoid 3 things here:
- Using `\` to create multi-line statements. If necessary, create an intermediate variable to hold the values temporarily.
- (Same as above): Accessing multiple times the same dict elements. Instead, let's capture the element in a variable (in this case, `structure`), and then use it as many times as needed. This will also help in the point mentioned above.
- Using `in` check on a dict element that you will end up getting from the dict if found. Instead, just directly take it using a `get` and then just check if that was `None` or not.

So, altogether, I would write this block of code like this:

```python3
structure = table.get('structure')
if structure:
    structure['structure'] = structure['structure'][:max_columns]
    structure['states'] = structure['states'][:max_columns]
```